ably/options.go

```diff
@@ -14,36 +15,74 @@ const (
 	ProtocolMsgPack = "msgpack"
 )
 
+var errInvalidKey = errors.New("invalid key format")
+
+var DefaultOptions = &ClientOptions{
+	RestEndpoint:     "rest.ably.io",
+	RealtimeEndpoint: "realtime.ably.io",
```

Comment:
Sorry, should have noticed before, but the option name could be considered a bit misleading as an end point is typically a FQDN i.e. "http://rest.ably.io" whereas what we actually allow is simply a custom host.  See https://github.com/ably/ably-java/blob/master/src/io/ably/types/ClientOptions.java#L61-L67 for the Java implementation.  Personally I prefer Rest & Realtime over the Java implementation, but can we use the name host instead please as this is more accurate and consistent, and we'll clean up the Java version too to match this?